clients/src/test/java/org/apache/kafka/clients/producer/internals/RecordAccumulatorTest.java

```diff
@@ -98,6 +99,54 @@ public void teardown() {
         this.metrics.close();
     }
 
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testDrainBatchesStarve() throws Exception {
```

Comment:
nice test

Reply:
I think we can change the test name to `testDrainBatches`, because we are now testing more scenarios.

Reply:
make sense